features/step_definitions/operator.rb

```diff
@@ -9,3 +9,141 @@
+    
+    step %Q/I run the :get client command with:/, table(%{
+      | resource | etcdcluster |                                                  
+    })
+    step %Q/the output should contain "No resources found"/
+    raise "Error find the etcdCluster resource in #{proj_name} project" unless @result[:success]
+    logger.info("### etcd operator: #{name} is installed successfully in #{proj_name} namespace")
+
+end
+
+Given /^etcdCluster "([^"]*)" is installed successfully in "([^"]*)" project$/ do | name, proj_name|
+  step %Q/I use the "#{proj_name}" project/
+  etcdCluster_yaml ||= "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/olm/etcd-cluster-template.yaml"
+  step %Q/I process and create:/, table(%{
+    | f | #{etcdCluster_yaml} |
+    | p | NAME=#{name}         |
+    | p | NAMESPACE=#{proj_name} |
+  })
+  raise "Error creating etcdCluster: #{name} in project: #{proj_name}" unless @result[:success]
+  logger.info("### etcdCluster: #{name} is installed successfully in #{proj_name} namespace")
+
+end
+
+Given /^etcdCluster "([^"]*)" is removed successfully from "([^"]*)" project$/ do | name, proj_name|
+  step %Q/I run the :delete client command with:/, table(%{
+    | object_type       | etcdcluster  | 
+    | object_name_or_id | #{name}      |
+    | n                 | #{proj_name} |      
+  })
+  step %Q/the output should contain "deleted"/
+  step %Q/I wait for the resource "etcdcluster" named "#{name}" to disappear within 180 seconds/
+  step %Q/I run the :get client command with:/, table(%{
+    | resource      | etcdcluster  | 
+    | resource_name | #{name}      |
+    | n             | #{proj_name} |      
+  })
+  step %Q/the output should contain "NotFound"/
+  logger.info("### etcdCluster: #{name} is removed successfully from #{proj_name} namespace")
+
+end
+
+Given /^The status of condition "([^"]*)" for "([^"]*)" operator is: (.+)$/ do | type, operator, status |
+  ensure_admin_tagged
+  expected_status = status
+
+  if type == "Available"
+    @result = admin.cli_exec(:get, resource: "clusteroperators", resource_name: operator, o: "jsonpath={.status.conditions[?(.type == \"Available\")].status}")
+    real_status = @result[:response]
+  elsif type == "Progressing"
+    @result = admin.cli_exec(:get, resource: "clusteroperators", resource_name: operator, o: "jsonpath={.status.conditions[?(.type == \"Progressing\")].status}")
+    real_status = @result[:response]
+  elsif type == "Degraded"
+    @result = admin.cli_exec(:get, resource: "clusteroperators", resource_name: operator, o: "jsonpath={.status.conditions[?(.type == \"Degraded\")].status}")
+    real_status = @result[:response]
+  elsif type == "Upgradeable"
+    @result = admin.cli_exec(:get, resource: "clusteroperators", resource_name: operator, o: "jsonpath={.status.conditions[?(.type == \"Upgradeable\")].status}")
+    real_status = @result[:response]
+  else
+    raise "Unknown condition type!"
+  end
```

Comment:
You can do here
```
@result = admin.cli_exec(:get, resource: "clusteroperators", resource_name: operator, o: "jsonpath={.status.conditions[?(.type == \"#{type}\")].status}")
```

This will avoid the need to repeat same command for each `type`. Even better:
```
ensure_admin_tagged
actual_status = cluster_operator(operator).condition(type: type, cached: false)['status']
unless status == actual_status
  raise "status of #{operator} condition #{type} is #{actual_status}"
end
```

And you can rename parameter in step name to be `expected_status` instead of creating new variable in step code.

Reply:
Great! Thanks!